File: src/event/impl/player/PlayerKickEvent.php
<?php

namespace pocketcloud\cloud\event\impl\player;

use pocketcloud\cloud\event\Cancelable;
use pocketcloud\cloud\event\CancelableTrait;
use pocketcloud\cloud\player\CloudPlayer;

class PlayerKickEvent extends PlayerEvent implements Cancelable {
    use CancelableTrait;

    public function __construct(
        CloudPlayer $player,
        protected readonly string $reason,
        protected readonly string $disconnectScreenMessage
    ) {
        parent::__construct($player);
    }

    public function getReason(): string {
        return $this->reason;
    }

    public function getDisconnectScreenMessage(): string {
        return $this->disconnectScreenMessage;
    }
}

File: cloud-core/src/main/java/de/pocketcloud/cloud/event/impl/player/PlayerKickEvent.java
package de.pocketcloud.cloud.event.impl.player;

import de.pocketcloud.cloud.event.Cancelable;
import de.pocketcloud.cloud.player.CloudPlayer;
import lombok.Getter;

/**
 * This event is called when you kick a CloudPlayer directly via the cloud with {@link CloudPlayer#kick()}
 */

@Getter
public class PlayerKickEvent extends PlayerEvent implements Cancelable {

    private final String reason;
    private final String disconnectScreenMessage;

    public PlayerKickEvent(CloudPlayer player, String reason, String disconnectScreenMessage) {
        super(player);
        this.reason = reason;
        this.disconnectScreenMessage = disconnectScreenMessage;
    }
}